src/decision/justification_strategy.h

```diff
@@ -144,32 +144,24 @@ class JustificationStrategy : public DecisionEngine
    */
   bool isDone() override;
   /**
-   * If skolem is null, notify this class that assertion is an (input)
-   * assertion, not corresponding to a skolem definition.
-   *
-   * If skolem is non-null, notify this class that lem is the skolem definition
-   * for skolem, which is a part of the current assertions.
-   */
-  void addAssertion(TNode lem, TNode skolem, bool isLemma) override;
-  /**
-   * Notify this class that the list of lemmas defs are now active in the
-   * current SAT context. This is triggered when a literal lit is sent to
-   * TheoryEngine that contains skolems we have yet to see in the current SAT
-   * context, where defs are the skolem definitions for each such skolem.
+   * Adds assertions lems to satisfy that persist in the user context.
+   * All input assertions and lemmas not marked "local" are added via this call.
    */
-  void notifyActiveSkolemDefs(std::vector<TNode>& defs) override;
+  void addAssertions(const std::vector<TNode>& lems) override;
   /**
-   * We need notification of active skolem definitions when our skolem
-   * relevance policy is JutificationSkolemRlvMode::ASSERT.
+   * Adds assertions lems to satisfy that persist in the SAT context.
+   * This is triggered when a literal lit is sent to TheoryEngine that contains
+   * skolems we have yet to see in the current SAT context, where lems are the
+   * skolem definitions for each such skolem.
    */
-  bool needsActiveSkolemDefs() const override;
+  void addLocalAssertions(const std::vector<TNode>& lems) override;
 
  private:
   /**
    * Helper method to insert assertions in `toProcess` to `d_assertions` or
    * `d_skolemAssertions` based on `useSkolemList`.
```

Comment:
Can you improve this documentation to indicate which case applies to which? Where is it added when `useSkolemList` is true?

Reply:
It should be renamed "local".

Reply:
Sure, but the main question was to clarify in which case it is added to `d_assertions`, when `local` is true?